paasta_tools/firewall.py

```diff
@@ -259,6 +261,72 @@ def _smartstack_rules(conf, soa_dir, synapse_service_dir):
         yield _yocalhost_rule(port, 'proxy_port ' + namespace)
 
 
+def _cidr_rules(conf):
+    for dep in conf.get_dependencies() or ():
+        cidr = dep.get('cidr')
+        port = dep.get('port')
+
+        if cidr is not None:
+            try:
+                network = ipaddress.IPv4Network(cidr)
+            except ipaddress.AddressValueError:
+                # TODO: surface this error to the user?
+                continue
+
+            if port is not None:
+                try:
+                    port = int(port)
+                except ValueError:
+                    # TODO: surface this error to the user?
+                    continue
+
+                if not 1 <= port <= 65535:
+                    # TODO: surface this error to the user?
+                    continue
+
+            # Set up an ip rule if no port, or a tcp/udp rule if there is a port
+            dst = f'{network.network_address.exploded}/{network.netmask}'
+            if port is None:
+                yield iptables.Rule(
+                    protocol='ip',
+                    src='0.0.0.0/0.0.0.0',
+                    dst=dst,
+                    target='ACCEPT',
+                    matches=(
+                        (
+                            'comment',
+                            (
+                                ('comment', (f'allow {network}:*',)),
```

Comment:
it's the network in CIDR notation, here are some examples from mesos1-uswest1cdevc:
```
-A PAASTA.robj-test-.187152c983 -d 169.229.226.12/32 -m comment --comment "allow 169.229.226.12/32:*" -j ACCEPT
-A PAASTA.robj-test-.187152c983 -d 169.229.226.128/25 -m comment --comment "allow 169.229.226.128/25:*" -j ACCEPT
-A PAASTA.robj-test-.187152c983 -d 169.229.226.0/25 -p udp -m comment --comment "allow 169.229.226.0/25:443" -m udp --dport 443 -j ACCEPT
-A PAASTA.robj-test-.187152c983 -d 169.229.226.0/25 -p tcp -m comment --comment "allow 169.229.226.0/25:443" -m tcp --dport 443 -j ACCEPT
-A PAASTA.robj-test-.187152c983 -d 169.229.226.0/25 -p udp -m comment --comment "allow 169.229.226.0/25:80" -m udp --dport 80 -j ACCEPT
-A PAASTA.robj-test-.187152c983 -d 169.229.226.0/25 -p tcp -m comment --comment "allow 169.229.226.0/25:80" -m tcp --dport 80 -j ACCEPT
```